File: src/main/java/com/dh/ondot/schedule/infra/exception/OdsayNoResultException.java
package com.dh.ondot.schedule.infra.exception;

import com.dh.ondot.core.exception.BadRequestException;

import static com.dh.ondot.core.exception.ErrorCode.ODSAY_NO_RESULT;

public class OdsayNoResultException extends BadRequestException {
    public OdsayNoResultException(String detail) {
        super(ODSAY_NO_RESULT.getMessage().formatted(detail));
    }
    @Override
    public String getErrorCode() {
        return ODSAY_NO_RESULT.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/exception/OdsayNoResultException.kt
package com.dh.ondot.schedule.infra.exception

import com.dh.ondot.core.exception.BadRequestException
import com.dh.ondot.core.exception.ErrorCode.ODSAY_NO_RESULT

class OdsayNoResultException(detail: String) :
    BadRequestException(ODSAY_NO_RESULT.message.format(detail)) {
    override val errorCode: String get() = ODSAY_NO_RESULT.name
}
